petab/parameter_mapping.py

```diff
@@ -63,36 +69,68 @@ def get_optimization_to_simulation_parameter_mapping(
         simulation_conditions = measurements.get_simulation_conditions(
             measurement_df)
 
-    mapping = []
-    for condition_ix, condition in simulation_conditions.iterrows():
-        cur_measurement_df = measurements.get_rows_for_condition(
-            measurement_df, condition)
+    simulation_parameter_ids = sbml.get_model_parameters(sbml_model)
 
-        if 'preequilibrationConditionId' not in condition \
-                or not isinstance(condition.preequilibrationConditionId, str) \
-                or not condition.preequilibrationConditionId:
-            preeq_map = {}
-        else:
-            preeq_map = get_parameter_mapping_for_condition(
-                condition_id=condition.preequilibrationConditionId,
-                is_preeq=True,
-                cur_measurement_df=cur_measurement_df,
-                condition_df=condition_df,
-                parameter_df=parameter_df, sbml_model=sbml_model,
-                warn_unmapped=warn_unmapped
-            )
+    num_processes = 1
+    if ENV_NUM_THREADS in os.environ:
+        num_processes = int(os.environ[ENV_NUM_THREADS])
+
+    from multiprocessing import Pool
+    pool = Pool(processes=num_processes)
```

Comment:
so now you perform the map in all cases, using multiprocessing. but that pickles the objects, right? so for 1 process, the default, you get some overhead?

Reply:
would multithreading be an option?

Reply:
Overhead, yes. Could add special treatment. Although for that case, not sure if the overhead is worse than adding more complexity. Multi-threading: Possibly.

Reply:
ok, I guess pickling only becomes a problem with large problems, but then you will wanna parallelize anyway, so should be fine

Reply:
threads and special handling of sequential -> done